apisix/plugins/limit-conn/limit-conn-redis.lua

```diff
@@ -0,0 +1,139 @@
+--
+-- Licensed to the Apache Software Foundation (ASF) under one or more
+-- contributor license agreements.  See the NOTICE file distributed with
+-- this work for additional information regarding copyright ownership.
+-- The ASF licenses this file to You under the Apache License, Version 2.0
+-- (the "License"); you may not use this file except in compliance with
+-- the License.  You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+local redis     = require("apisix.utils.redis")
+local core          = require("apisix.core")
+local assert        = assert
+local math          = require "math"
+local floor         = math.floor
+local ngx_timer_at  = ngx.timer.at
+
+local setmetatable  = setmetatable
+
+
+local _M = {version = 0.1}
+
+
+local mt = {
+    __index = _M
+}
+
+function _M.new(plugin_name, conf, max, burst, default_conn_delay)
+
+    local self = {
+        conf = conf,
+        plugin_name = plugin_name,
+        burst = burst,
+        max = max + 0,    -- just to ensure the param is good
+        unit_delay = default_conn_delay,
+    }
+    return setmetatable(self, mt)
+end
+
+
+function _M.incoming(self, key, commit)
```

Comment:
the `incoming` function is same to same in `redis-cluster`. How do you plan to unify this?

Reply:
Perhaps we can create a util.lua and pass red_cli as a parameter to avoid duplicating the code for the incoming and leaving functions?